plugins/optimization-detective/tests/storage/test-class-od-url-metrics-post-type.php

```diff
@@ -323,6 +323,78 @@ public function test_delete_stale_posts(): void {
 		$this->assertInstanceOf( WP_Post::class, get_post( $old_generic_post ), 'Expected old generic post to not have been deleted.' );
 		$this->assertInstanceOf( WP_Post::class, get_post( $new_url_metrics_post ), 'Expected new URL Metrics post to not have been deleted.' );
 		$this->assertNull( get_post( $old_url_metrics_post ), 'Expected old URL Metrics post to have been deleted.' );
+
+		// Remove the URL Metrics post older than 6 months.
+		add_filter(
+			'od_url_metric_garbage_collection_ttl',
+			static function (): int {
+				return 6 * MONTH_IN_SECONDS;
+			}
+		);
+
+		// Update timestamp to 6 months older.
+		$new_stale_timestamp_gmt = gmdate( 'Y-m-d H:i:s', time() - ( 6 * MONTH_IN_SECONDS * HOUR_IN_SECONDS ) );
```

Comment:
Oh, why is this multiplying `MONTH_IN_SECONDS * HOUR_IN_SECONDS`? Shouldn't this rather be rather one second longer than the expiration age of `6 * MONTH_IN_SECONDS`?:

```suggestion
		$new_stale_timestamp_gmt = gmdate( 'Y-m-d H:i:s', time() - ( 6 * MONTH_IN_SECONDS + 1 ) );
```

Reply:
> Shouldn't this rather be rather one second longer than the expiration age of 6 * MONTH_IN_SECONDS

Yes it could be, but I thought to carry the same from the above where HOURS_IN_SECONDS was used.

I would change to `+ 1` rather. but not sure if it could make much difference IMO?

Reply:
I'm not sure I understand. What is `MONTH_IN_SECONDS * HOUR_IN_SECONDS` supposed to mean? That's like saying 3600 months of seconds. Multiplying seconds by other seconds doesn't make sense to me.

Reply:
This makes sense, but I added because it was previously added here in the test - https://github.com/WordPress/performance/blob/48db4baaa8cf5b4459597684a00c6b2af273f5b2/plugins/optimization-detective/tests/storage/test-class-od-url-metrics-post-type.php#L291, I thought to carry forward the same.

So to make more consistent I have added that.

Reply:
Ah, but that is quite different. That's subtracting one unit of seconds from another unit of seconds, so the resulting unit is seconds. By multiplying two units of seconds together then you kinda get a strange unit: seconds².

Reply:
> Ah, but that is quite different. That's subtracting one unit of seconds from another unit of seconds, so the resulting unit is seconds. By multiplying two units of seconds together then you kinda get a strange unit: seconds².

Yep Gotcha!